arch/Kconfig

```diff
@@ -259,9 +259,17 @@ config ARCH_NEED_ADDRENV_MAPPING
 	bool
 	default n
 
+config ARCH_HAVE_EXTRA_HEAPS
+	bool
+	default n
+	---help---
+		Special memory regions used as separate heaps
+
 config ARCH_HAVE_TEXT_HEAP
-	bool "Special memory region for dynamic code loading"
+	bool
 	default n
```

Comment:
should we select ARCH_HAVE_EXTRA_HEAPS?

Reply:
This won't work well with SoCs that don't have an extra heap but just offset the buffer gotten from simple malloc.  Like [CXD56](https://github.com/apache/incubator-nuttx/blob/master/arch/arm/src/cxd56xx/cxd56_textheap.c#L50-L80).  Part of what  I did here is to be able to remove that `up_textheap_init` function when it's not required.  (The term `textheap` could be a bit misleading here though).